### DISCUSSION TITLE
The selected instance type is not available with the current billing plan. ON MAC M1

### DISCUSSION THREAD

**Discussion by anchous676** [codemagic-ci-cd/codemagic-docs#2728] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

workflows:
  ios-native-quick-start:
    name: iOS Native
    max_build_duration: 120
    instance_type: mac_mini_m1
    integrations:
      app_store_connect: Codemagic
    environment:
      ios_signing:
        distribution_type: app_store
        bundle_identifier: bub.bubble.shot
      vars:
        BUNDLE_ID: "bub.bubble.shot"
        XCODE_WORKSPACE: "bubBubble Shot.xcworkspace"
        XCODE_SCHEME: "bubBubble Shot"
        APP_STORE_APP_ID: "--------"
      xcode: latest # <-- set to specific version e.g. 14.3, 15.0 to avoid unexpected updates.
      cocoapods: default
    scripts:
      - name: Install CocoaPods dependencies
        script: | 
          pod install
      - name: Set up provisioning profiles settings on Xcode project
        script: xcode-project use-profiles
      - name: Increment build number
        script: | 
          cd $CM_BUILD_DIR
          LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number "$APP_STORE_APPLE_ID")
          agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
      - name: Build ipa for distribution
        script: | 
          xcode-project build-ipa \
            --workspace "$CM_BUILD_DIR/$XCODE_WORKSPACE" \
            --scheme "$XCODE_SCHEME"
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - $HOME/Library/Developer/Xcode/DerivedData//Build//*.app
      - $HOME/Library/Developer/Xcode/DerivedData//Build//*.dSYM
    publishing:
      email:
        recipients:
          - user_1@example.com
          - user_2@example.com
        notify:
          success: true
          failure: false
      app_store_connect:
        auth: integration

        # Configuration related to TestFlight (optional)
        # Note: This action is performed during post-processing.
        submit_to_testflight: true
        beta_groups: # Specify the names of beta tester groups that will get access to the build once it has passed beta review.
          - group name 1
          - group name 2

        # Configuration related to App Store (optional)
        # Note: This action is performed during post-processing.
        submit_to_app_store: false

### Expected results

Should start a build (I changed the Apple ID, and BundleID) on the actual yaml they are correct.

### Actual results

I immediately get an error “The selected instance type is not available with the current billing plan.” as if I'm using a paid instance
![image](https://github.com/user-attachments/assets/840ee287-93a6-468b-91ec-eebec7cf0a28)

### Build id (optional)

66e011212d5599513d424d63

**Comment by helinanever** [comment, marked as answer]
Hi @anchous676 , we believe this was a temporary issue that has been fixed now. Please try again. Your build should now run on M2, see this [accouncement](https://github.com/orgs/codemagic-ci-cd/discussions/2659).

**Reply by flodaniel** [reply]
Just came here as I am also getting this error on my builds. They worked a couple hours ago.

Edit: I found this paragraph in the announcement

> While the existing configuration will continue to work, we advise you to update the mac_mini_m1 instance type to mac_mini_m2 in codemagic.yaml. Otherwise, you’ll be seeing a message about using a deprecated instance type in build logs.

After changing to `mac_mini_m2` it works again. Looks like the compatibility with the old m1 instance_type is broken.

**Reply by helinanever** [reply]
Thanks for reporting this. We needed to make another adjustment, but this should be fixed now.
